Guild Bank Update

Drummss opened the pull request:
This fixes an issue where the command would error if the user didn't already have a bank account for whatever guild they are depositing into.

- Added BankAccountService.
- Added BankAccountService tests.
- Cleanup guild bank commands.

Drummss committed:
Fix guildbank deposit

- Added BankAccountService.
- Added BankAccountService tests.
- Cleanup guildbank commands.

velddev commented on src/Miki/Services/BankAccountService.cs:
```diff
@@ -0,0 +1,82 @@
+
+    public interface IBankAccountService : IDisposable
+    {
+        ValueTask<BankAccount> GetOrCreateAccountAsync(long userId, long guildId);
```
This should be an extension method, consider moving this to a get and create for better reusability

velddev commented on src/Miki/Services/BankAccountService.cs:
```diff
@@ -0,0 +1,82 @@
+
+        ValueTask<BankAccount> DepositAsync(long userId, long guildId, int amount);
+
+        ValueTask<BankAccount> WithdrawAsync(long userId, long guildId, int amount);
```
Should we create a guilduser type to reuse for these kinds of situations to avoid having to pass multiple ids?

velddev commented on src/Miki/Services/BankAccountService.cs:
```diff
@@ -0,0 +1,82 @@
+
+        /// <inheritdoc />
+        public async ValueTask<BankAccount> WithdrawAsync(long userId, long guildId, int amount)
+            => throw new NotImplementedException();
```
Add a to-do here please

velddev reviewed (approved):
Looking good, I left a few suggestions and stuff here and there, but it's not required.

Drummss committed:
BankAccountService Polish

- Added CreateAccountAsync and GetAccountAsync to BankAccountService.
- Moved GetOrCreateAccountAsync to ServiceExtensions as
  GetOrCreatBankAccountAsync.
- Updated relative code to accommodate the other changes.

velddev commented on src/Miki/Extensions/ServiceExtensions.cs:
```diff
@@ -18,10 +18,26 @@ public static class ServiceExtensions
                 return await service.GetUserAsync((long)user.Id);
             }
-            catch(UserNullException)
+            catch (UserNullException)
```
nit: random change?

Drummss replied:
Automatically formatted

velddev commented on src/Miki/Services/BankAccountService.cs:
```diff
@@ -0,0 +1,104 @@
+
+        // TODO: Add withdraw capability.
+        /// <inheritdoc />
+        public async ValueTask<BankAccount> WithdrawAsync(AccountDetails accountDetails, int amount)
```
nit: remove async from this to avoid warnings

velddev commented on src/Miki/resources/locales/eng.json:
```diff
@@ -647,6 +647,7 @@
   "miki_command_usage_setnotifications": ">setnotifications [-type:(rewards only|all|none)] [-g:true]",
   "guildhouse_buy": "For your next upgrade. Your guild needs to pay {0} mekos! use `>guildupgrade house` to purchase it.",
   "error_user_null": "User not found!",
+  "error_bank_account_null": "Bank account doesn't exist!",
```
the user doesn't know what a bank account is, this wouldn't make much sense to read

Drummss committed:
Fix NITs
